services/QuillLMS/client/app/bundles/Teacher/startup/CmsDistrictIndexAppClient.tsx

```diff
@@ -0,0 +1,236 @@
+        Cell: row => Number(row.original.total_schools),
+      }, {
+        Header: "Students",
+        accessor: 'total_students',
+        resizable: false,
+        minWidth: 80,
+        Cell: row => Number(row.original.total_students),
+      }, {
+        Header: "Edit",
+        accessor: 'edit',
+        resizable: false,
+        minWidth: 60,
+        Cell: (row) => {
+          return <a href={`${process.env.DEFAULT_URL}/cms/districts/${row.original.id}`}>Edit</a>
+        }
+      }
+    ];
+  }
+
+  setSort = newSorted => {
+    const { query } = this.state
+
+    const sort = newSorted[0].id
+    const sort_direction = newSorted[0].desc ? 'desc' : 'asc'
+    if (sort !== query.sort || sort_direction !== query.sort_direction) {
+      const newState = { ...this.state}
+      newState.query.sort = sort
+      newState.query.sort_direction = sort_direction
+      this.setState(newState, this.search)
+    }
+  };
+
+  search = (e) => {
+    const { query } = this.state
+
+    e ? e.preventDefault() : null
+    this.setState({loading: true})
+    const link = `${process.env.DEFAULT_URL}/cms/districts/search`
+    const data = new FormData();
+    data.append('district', '')
+    Object.keys(query).forEach((k) => {
+      data.append(k, query[k])
+    })
+    fetch(link, {
+      method: 'POST',
+      mode: 'cors',
+      credentials: 'include',
+      body: data,
+      headers: {
+        'X-CSRF-Token': getAuthToken()
+      }
+    }).then((response) => {
+      if (!response.ok) {
+        throw Error(response.statusText);
+      }
+      return response.json();
+    }).then((response) => {
+      this.setState({ data: response.districtSearchQueryResults, numberOfPages: response.numberOfPages, loading: false })
+    }).catch((error) => {
+      // to do, use Sentry to capture error
+    })
+  };
+
+  submitPageForm = e => {
+    this.updatePage(e.target.page.value)
+  };
+
+  updateField(e, key) {
+    const value = e.target.value
+    const newState = { ...this.state}
+    newState.query[key] = value
+    this.setState(newState)
+  }
+
+  updatePage = i => {
+    const newState = { ...this.state}
+    newState.query.page = i
+    this.setState(newState, this.search)
+  };
+
+  renderPageSelector() {
+    const { query, numberOfPages } = this.state
+
+    const currentPage = query.page || 1
+    const totalPages = numberOfPages || 1
+    return (
+      <div style={{display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
+        <a onClick={() => this.updatePage(1)}>First</a>
+        <form onSubmit={this.submitPageForm}>
+          <input defaultValue={currentPage} name='page' /><span>of {totalPages}</span>
+        </form>
+        <a onClick={() => this.updatePage(totalPages)}>Last</a>
+      </div>
+    )
+  }
+
+  renderTableOrLoading() {
+    const { loading, data, columns, query } = this.state
+
+    if (loading) {
+      return <LoadingIndicator />
+    } else if (data && data.length) {
```

Comment:
You don't need else if since the above returns.
```suggestion
    } 
    
    if (data && data.length) {
```